### PR TITLE
CSS modules: Calendar, DatePicker

### PR DESCRIPTION
**Pull request by dzearing**
This update patches the scss to use css modules for these components:

Calendar
DatePicker

Note: this PR is for the `module-css-updates` branch. We'll merge all changes there, publish a beta build, test some projects with that, then merge into master and publish after we've done some internal testing.

To make the PR easier to read, you can disable whitespace on the diff by adding the ?w=1 parameter to the PR url:

https://github.com/OfficeDev/office-ui-fabric-react/pull/1049/files?w=1

### CODE REVIEW

File: common/changes/people-picker-button-padding_2017-02-16-17-28.json [hunk 1]
```diff
@@ -0,0 +1,15 @@
+{
+  "changes": [
+    {
+      "packageName": "office-ui-fabric-react",
+      "comment": "Return suggested items to original padding values, add display block to fix following space due to inline-block",
```
Comment (mikewheaton) [inline common/changes/people-picker-button-padding_2017-02-16-17-28.json]: Which component does this affect?

File: common/changes/users-rsemenov-1018-messagebar-focus-order_2017-02-16-20-20.json [hunk 2]
```diff
@@ -0,0 +1,10 @@
+{
+  "changes": [
+    {
+      "packageName": "office-ui-fabric-react",
+      "comment": "Fix keyboard focus order in MessageBar",
```
Comment (mikewheaton) [inline common/changes/users-rsemenov-1018-messagebar-focus-order_2017-02-16-20-20.json]: "MessageBar: Fix keyboard focus order"

File: common/changes/alert-dialog_2017-02-20-10-12.json [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+      "packageName": "@uifabric/utilities",
+      "type": "none"
+    },
+    {
+      "comment": "",
+      "packageName": "@uifabric/example-app-base",
+      "type": "none"
+    }
+  ],
+  "email": "manishda@microsoft.com"
```
Comment (mikewheaton) [inline common/changes/alert-dialog_2017-02-20-10-12.json]: Why are so many changes in this PR? Was expecting it to just the switch to CSS modules.

File: packages/office-ui-fabric-react/src/components/Calendar/Calendar.tsx [hunk 4]
```diff
@@ -73,11 +74,18 @@ export class Calendar extends BaseComponent<ICalendarProps, ICalendarState> impl
...
-      <div className={ rootClass } ref='root'>
-        <div className={ 'ms-DatePicker-picker ms-DatePicker-picker--opened ms-DatePicker-picker--focused ' + (this.props.isMonthPickerVisible ? 'is-monthPickerVisible' : '') } >
-          <div className='ms-DatePicker-holder' onKeyDown={ this._onDatePickerPopupKeyDown }>
-            <div className='ms-DatePicker-frame'>
-              <div className='ms-DatePicker-wrap'>
+      <div className={ css(rootClass, styles.root) } ref='root'>
+        <div className={ css(
+          'ms-DatePicker-picker ms-DatePicker-picker--opened ms-DatePicker-picker--focused ',
+          styles.picker,
+          styles.pickerOpened,
```
Comment (mikewheaton) [inline packages/office-ui-fabric-react/src/components/Calendar/Calendar.tsx]: `pickerIsOpened`

File: packages/office-ui-fabric-react/src/components/Calendar/Calendar.tsx [hunk 5]
```diff
@@ -73,11 +74,18 @@ export class Calendar extends BaseComponent<ICalendarProps, ICalendarState> impl
...
+          styles.pickerFocused,
```
Comment (mikewheaton) [inline packages/office-ui-fabric-react/src/components/Calendar/Calendar.tsx]: `pickerIsFocused`

File: packages/office-ui-fabric-react/src/components/Calendar/Calendar.tsx [hunk 6]
```diff
@@ -73,11 +74,18 @@ export class Calendar extends BaseComponent<ICalendarProps, ICalendarState> impl
...
+          this.props.onDismiss && 'is-monthPickerVisible',
+          this.props.onDismiss && styles.isMonthPickerVisible
```
Comment (mikewheaton) [inline packages/office-ui-fabric-react/src/components/Calendar/Calendar.tsx]: `pickerIsMonthPickerVisible`

File: packages/office-ui-fabric-react/src/components/Calendar/CalendarDay.tsx [hunk 7]
```diff
@@ -108,12 +120,19 @@ export class CalendarDay extends BaseComponent<ICalendarDayProps, ICalendarDaySt
...
-                        className={ css('ms-DatePicker-day', {
-                          'ms-DatePicker-day--infocus': day.isInMonth,
-                          'ms-DatePicker-day--outfocus': !day.isInMonth,
-                          'ms-DatePicker-day--today': day.isToday,
-                          'ms-DatePicker-day--highlighted': day.isSelected
-                        }) }
+                        className={ css(
+                          'ms-DatePicker-day',
+                          styles.day,
+                          day.isInMonth && styles.dayInfocus,
```
Comment (mikewheaton) [inline packages/office-ui-fabric-react/src/components/Calendar/CalendarDay.tsx]: `dayIsInMonth` or `dayIsFocused`

File: packages/office-ui-fabric-react/src/components/Calendar/CalendarDay.tsx [hunk 8]
```diff
@@ -108,12 +120,19 @@ export class CalendarDay extends BaseComponent<ICalendarDayProps, ICalendarDaySt
...
+                          !day.isInMonth && styles.dayOutfocus,
+                          day.isToday && styles.dayToday,
```
Comment (mikewheaton) [inline packages/office-ui-fabric-react/src/components/Calendar/CalendarDay.tsx]: Can we keep the "is" in the state? E.g. `dayIsToday`

File: packages/office-ui-fabric-react/src/components/Calendar/CalendarDay.tsx [hunk 9]
```diff
@@ -108,12 +120,19 @@ export class CalendarDay extends BaseComponent<ICalendarDayProps, ICalendarDaySt
...
+                          day.isSelected && styles.dayHighlighted,
```
Comment (mikewheaton) [inline packages/office-ui-fabric-react/src/components/Calendar/CalendarDay.tsx]: `dayIsSelected` or `dayIsHighlighted`

File: packages/office-ui-fabric-react/src/components/pickers/Suggestions/Suggestions.scss [hunk 10]
```diff
@@ -24,6 +24,8 @@
       height: auto;
       width: 100%;
       @include text-align(left);
+      padding: 4px 20px 6px;
```
Comment (mikewheaton) [inline packages/office-ui-fabric-react/src/components/pickers/Suggestions/Suggestions.scss]: Nit: It's often confusing when there are three values, `padding: 4px 20px 6px 20px` can prevent future problems.

File: common/changes/alert-dialog_2017-02-20-10-12.json (hunk 3 above)
Comment (micahgodbolt) [inline common/changes/alert-dialog_2017-02-20-10-12.json]: I was about to ask the same. type is now size?

File: packages/office-ui-fabric-react/src/components/Calendar/Calendar.scss [hunk 11]
```diff
@@ -9,524 +9,522 @@
...
+.dayIsHighlighted:hover,
+.pickerIsFocused .dayIsHighlighted {
+  cursor: pointer;
+  color: $ms-color-white;
+  background: $ms-color-themePrimary;
+}
+
+
+// Disabled and highlighted dates.
+.dayIsHighlighted.dayDisabled,
```
Comment (mikewheaton) [inline packages/office-ui-fabric-react/src/components/Calendar/Calendar.scss]: Is this supposed to be `dayIsDisabled`?

File: packages/office-ui-fabric-react/src/components/Calendar/Calendar.scss [hunk 12]
```diff
@@ -9,524 +9,522 @@
...
 
-
-  // Hovered date picker items.
-  .ms-DatePicker-day--infocus:hover,
-  .ms-DatePicker-day--outfocus:hover {
-    cursor: pointer;
-    color: $ms-color-black;
-    background: $ms-color-neutralLight;
+// State: The picker is showing the year components.
+.root.isPickingYears {
```
Comment (mikewheaton) [inline packages/office-ui-fabric-react/src/components/Calendar/Calendar.scss]: `.root.rootIsPickingYears`

File: packages/office-ui-fabric-react/src/components/Calendar/Calendar.scss [hunk 13]
```diff
@@ -9,524 +9,522 @@
...
+      padding: 0 10px;
+      @include right(10px);
+      @include text-align(right);
+      top: 199px;
+      width: 210px;
+    }
+    // State: The picker is showing the year components.
+    // On larger screens the calendar will remain and the years
+    // will replace the months.
+    .root.isPickingYears {
```
Comment (mikewheaton) [inline packages/office-ui-fabric-react/src/components/Calendar/Calendar.scss]: `.root.rootIsPickingYears`

File: packages/office-ui-fabric-react/src/components/Calendar/Calendar.scss [hunk 14]
```diff
@@ -9,524 +9,522 @@
...
 
-  // On smaller screens the month button toggles to the picking months state.
-  @media (max-width: 459px) {
+// On smaller screens the month button toggles to the picking months state.
+@media (max-width: 459px) {
 
-    // State: The picker is showing the month components.
-    .ms-DatePicker.is-pickingMonths {
+  // State: The picker is showing the month components.
+  .root.isPickingMonths {
```
Comment (mikewheaton) [inline packages/office-ui-fabric-react/src/components/Calendar/Calendar.scss]: `.root.rootIsPickingMonths`

File: packages/office-ui-fabric-react/src/components/DatePicker/DatePicker.scss [hunk 15]
```diff
@@ -15,32 +15,31 @@
...
-  .ms-DatePicker-event--without-label {
-    @include ms-DatePicker-event();
-    top: 7px;
+  input[readonly] {
+    cursor: pointer;
   }
 }
+
+// Insert a calendar icon over the date field.
+.eventWithLabel {
```
Comment (mikewheaton) [inline packages/office-ui-fabric-react/src/components/DatePicker/DatePicker.scss]: Maybe `.eventHasLabel` and `.eventHasNoLabel`. Doesn't sound as good but sticking to "is" or "has" for state/modifier classes will help them stand out.

**Pull request merged**